UnityAds/UnityFullscreenAdAdapter.m

```diff
@@ -0,0 +1,184 @@
+//
+
+#import "UnityFullscreenAdAdapter.h"
+#import "UnityAdsInstanceMediationSettings.h"
+#import "UnityAdsAdapterConfiguration.h"
+#import "UnityRouter.h"
+#import "NSError+UnityAdAdapter.h"
+#if __has_include("MoPub.h")
+    #import "MPAdAdapterError.h"
+    #import "MPReward.h"
+    #import "MPLogging.h"
+#endif
+
+static NSString *const kMPUnityRewardedVideoGameId = @"gameId";
+static NSString *const kUnityAdsOptionPlacementIdKey = @"placementId";
+static NSString *const kUnityAdsOptionZoneIdKey = @"zoneId";
+
+@interface UnityFullscreenAdAdapter ()
+
+@property (nonatomic, copy) NSString *placementId;
+@property (nonatomic, strong) NSString *objectId;
+
+@end
+
+@implementation UnityFullscreenAdAdapter
+@dynamic delegate;
+@dynamic localExtras;
+@dynamic hasAdAvailable;
+
+- (NSString *)getAdNetworkId {
+    return (self.placementId != nil) ? self.placementId : @"";
+}
+
+- (NSString *)adType {
+    return nil;
+ }
+
+#pragma mark - MPFullscreenAdAdapter Override
+
+- (BOOL)enableAutomaticImpressionAndClickTracking {
+    return NO;
+}
+
+- (BOOL)isRewardExpected {
+    return NO;
+}
+
+- (BOOL)hasAdAvailable {
+    return _placementId != nil;
+}
+
+- (void)requestAdWithAdapterInfo:(NSDictionary *)info adMarkup:(NSString *)adMarkup
+{
+    NSString *gameId = [info objectForKey:kMPUnityRewardedVideoGameId];
+   
+    if (gameId == nil) {
+        NSError *error = [NSError errorWithAdAdapterErrorCode: MPAdAdapterErrorCodeInvalidAdapter
+                                                  description: @"Custom event class data did not contain gameId."
+                                                   suggestion: @"Update your MoPub custom event class data to contain a valid Unity Ads gameId."];
+        MPLogAdEvent([MPLogEvent adLoadFailedForAdapter:NSStringFromClass(self.class) error:error], [self getAdNetworkId]);
+        [self.delegate fullscreenAdAdapter:self didFailToLoadAdWithError:error];
+        return;
+    }
+
+    // Only need to cache game ID for SDK initialization
+    [UnityAdsAdapterConfiguration updateInitializationParameters:info];
+
+    NSString *placementId = [info objectForKey:kUnityAdsOptionPlacementIdKey];
+    if (placementId == nil) {
+        placementId = [info objectForKey:kUnityAdsOptionZoneIdKey];
+    }
+
+    if (placementId == nil) {
+        NSError *error = [NSError errorWithAdAdapterErrorCode: MPAdAdapterErrorCodeInvalidAdapter
+                                                  description: @"Custom event class data did not contain placementId."
+                                                   suggestion: @"Update your MoPub custom event class data to contain a valid Unity Ads placementId."];
+        MPLogAdEvent([MPLogEvent adLoadFailedForAdapter:NSStringFromClass(self.class) error:error], placementId);
+        [self.delegate fullscreenAdAdapter:self didFailToLoadAdWithError:error];
+        return;
+    }
+    
+    UADSLoadOptions *options = [UADSLoadOptions new];
+    if (adMarkup != nil) {
```

Comment:
I've removed check for isInitialized because we want to handle all errors on webview side, so I think the same logic should be followed in other places as well. 

Also, adMarkup == nil  is a valid use case (not header bidding), objectId was just generated, so could not be nil.